pkg/controller/jobframework/reconciler.go

```diff
@@ -596,8 +596,8 @@ func (r *JobReconciler) ensureOneWorkload(ctx context.Context, job GenericJob, o
 		toDelete = toDelete[1:]
 	}
 
-	// If there is no matching workload and the job is running, suspend it.
-	if match == nil && !job.IsSuspended() {
+	// If there is no matching workload suspend job.
+	if match == nil {
```

Comment:
The main problem is that a "half stopped" will have the "running"  pod set info making it not equivalent to its workload , therefore the problematic part of this condition if  is `match==nil` not `!job.IsSuspended()`.